### PR TITLE
Internal Updates - Bust Cache for Listings and Add Note for Credits

### PR DESCRIPTION
**Pull request by jessleenyc**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [x] Bug Fix
- [ ] Documentation Update

## Description

I accidentally pushed this to the same branch! They're somewhat related though.

- Busts cache and reindexes a listing when it's updated from internal/listings
- Required an admin to leave a reason for removing/adding credits to individuals/organizations.

### CODE REVIEW

File: app/controllers/internal/classified_listings_controller.rb [hunk 1]
```diff
@@ -26,4 +27,12 @@ def destroy
   def listing_params
     params.require(:classified_listing).permit(:published, :body_markdown, :title, :category, :tag_list)
   end
+
+  def reindex_and_bust_cache
+    @classified_listing.index!
+    cb = CacheBuster.new
```
Comment (benhalpern) [inline app/controllers/internal/classified_listings_controller.rb]: Nice re-use of the object. I always forget to do this.

File: app/controllers/internal/classified_listings_controller.rb [hunk 2]
```diff
@@ -26,4 +27,12 @@ def destroy
   def listing_params
     params.require(:classified_listing).permit(:published, :body_markdown, :title, :category, :tag_list)
   end
+
+  def reindex_and_bust_cache
+    @classified_listing.index!
```
Comment (benhalpern) [inline app/controllers/internal/classified_listings_controller.rb]:
```suggestion
    @classified_listing.index! if @classified_listing.published
```

Should only index if published.

Comment (maestromac) [reply app/controllers/internal/classified_listings_controller.rb]: can users unpublish their listings? if they can, wouldn’t we want to re-index (or remove) it?

Comment (benhalpern) [reply app/controllers/internal/classified_listings_controller.rb]: Yup. We already do that in the controller.

In general this whole implementation needs to be refactored into some objects so we don't have to repeat the logic all over the place.

**Pull request merged**